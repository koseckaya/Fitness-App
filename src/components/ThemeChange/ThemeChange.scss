@import "src/styles/vars";

.scl-theme-change {
  box-sizing: border-box;
}
