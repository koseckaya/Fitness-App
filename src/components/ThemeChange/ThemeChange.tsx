import { FC } from "react";
import { Toggle } from "../Toggle";
import { ThemeContext } from "../utils/contexts";
import { themes } from "../utils/contexts/ThemeContext";

export type Props = {};

const ThemeChange: FC<Props> = ({}: Props) => (
    <ThemeContext.Consumer>
        {({ theme, setTheme }) => (
            <Toggle
                onChange={() => {
                    if (theme === themes.light && setTheme) setTheme(themes.dark);
                    if (theme === themes.dark && setTheme) setTheme(themes.light);
                }}
                value={theme === themes.dark}
            />
        )}
    </ThemeContext.Consumer>
);

export default ThemeChange;
