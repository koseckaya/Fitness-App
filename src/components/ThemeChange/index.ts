export { default as ThemeChange } from './ThemeChange';
export type { Props as ThemeChangeProps } from './ThemeChange';
