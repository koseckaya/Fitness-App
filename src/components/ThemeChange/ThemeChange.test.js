import React from 'react';
import renderer from 'react-test-renderer';
import ThemeChange from './ThemeChange';

it('renders correctly', () => {
    const tree = renderer
        .create(<ThemeChange></ThemeChange>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});