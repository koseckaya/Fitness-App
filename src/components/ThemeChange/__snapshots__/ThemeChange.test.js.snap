// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="toggle-container"
>
  <label
    className="switch"
    htmlFor="toggler"
  >
    <input
      checked={false}
      className="toggle-input"
      id="toggler"
      onClick={[Function]}
      readOnly={true}
      type="checkbox"
    />
    <span
      className="toggle-slider"
    />
    <span
      className="toggle-wave"
    />
  </label>
</div>
`;
